src/readfish/_utils.py

```diff
@@ -449,6 +453,48 @@ def get_device(
             return position
     raise ValueError(f"Could not find device {device!r}")
 
+def get_version(
```

Comment:
I would move this into `compatibility.py`, as it's compatibility related?

Reply:
Then you don't have to import the only things in `comapability` into this file

Reply:
also if you need to immediately rename it `as compatible_versions`, could it be called that to begin with?

Reply:
The docstring here says `Get the version of MinKNOW`

which tells me zero about what is actually happening underneath with all the conditionals and loops - needs expanding on

Reply:
Needs doctests as well!

Reply:
good point on renaming.